### ISSUE TITLE
Does not work with Angular 5 release - Http deprecated

### ISSUE THREAD

**Issue by jjmerri** [Tgure/ngx-loggly-logger#8]
The project needs to be updated to not use Http since it is deprecated. Cannot run with angular 5

**Comment by Tgure** [comment]
@jjmerri I have not updated my projects yet to 5, but using a angular 5 starter I made changes to this library to work with HttpClient and tested as working for me.  Please try version 5 and let me know.

**Comment by jjmerri** [comment]
@Tgure Thank you for jumping on this so quick! It is working great. Please close out the issue.

I have one unrelated question for you. I had copied loggly.service.ts to my project to make the changes myself but I kept getting 405 errors because my browser was forcing an OPTIONS call to Loggly which it doesn't support. I think it is related to CORS. My question is: Did you do anything to specifically to skip the OPTIONS call and just do a POST to Loggly?

**Comment by Tgure** [comment]
@jjmerri There should not be an options call since the header should be text/plain.  Do you have any global headers set for other calls?

**Comment by jjmerri** [comment]
@Tgure I copied your new file and it worked, so something I did when I converted it myself caused the problem. Below is how I made the POST which apparently caused the OPTIONS request. I have no idea why it would cause that but how you have it works fine. I got rid of the map() call too because I thought it wasn't supported anymore but I dont think that would cause it either.

        data.sessionId = this.session_id;
        let headers: any = new HttpHeaders({'Content-Type': 'text/plain'});
        return this._http.post(this.inputUrl, data, headers)

EDIT:
Ugh I see what happend now. I needed to do this: 
        return this._http.post(this.inputUrl, data, { headers: headers })
